app/src/components/ExportPane.js

```diff
@@ -2,55 +2,56 @@ import React, { useMemo } from 'react';
 
 import s from './ExportPane.module.css';
 import cs from 'clsx';
+import { Divider, Heading } from '@chakra-ui/react';
 
-import displayNames from '../displayNames';
-
-function ResultRow({ result }) {
-  const attrs = useMemo(() => {
-    const a = [];
-
-    (result.Attributes || []).forEach((attr) => {
-      a.push([displayNames[attr.Type], attr.Text]);
-    });
-
-    return a;
-  }, [result]);
-
-  return (
-    <div className={s.result}>
-      <h4>{result.Text}</h4>
-
-      <dl>
-        {attrs.map(([key, value]) => (
-          <React.Fragment key={key}>
-            <dt>{key}</dt>
-            <dd>{value}</dd>
-          </React.Fragment>
-        ))}
-      </dl>
-    </div>
-  );
+function conceptScoreSort(conceptArray) {
+  conceptArray.sort(function (concept1, concept2) {
+    return concept2.Score - concept1.Score;
+  });
+  return conceptArray;
 }
 
-function ResultTable({ results, category }) {
+function CategorySummary({ results, category }) {
```

Comment:
There are a few issues with this that the below implementation would fix:
```jsx
const conceptScoreSort = (conceptArray) =>
  [...conceptArray].sort((concept1, concept2) => concept2.Score - concept1.Score);

const getFormattedResult = (category, filteredResults) => {
  const isMedicalCondition = category === 'MEDICAL_CONDITION';
  const isMedication = category === 'MEDICATION';

  if (isMedicalCondition || isMedication) {
    const conceptProperty = isMedicalCondition ? 'ICD10CMConcepts' : 'RxNormConcepts';

    return filteredResults
      .map((result) => {
        const text = result.Text;

        const concepts = result[conceptProperty];

        if (!concepts) return text;

        // this should really be based on which concept was picked in the dropdown, but we don't have that state yet
        const chosenConcept = conceptScoreSort(concepts)[0];

        return `${text}|${chosenConcept.Code}|${chosenConcept.Description}`;
      })
      .join('\n');
  }

  return filteredResults.length > 0
    ? filteredResults
        .map(({ Text, Attributes }) => `${Text}${Attributes?.map((key) => `|${key.Text}`).join('') ?? ''}`)
        .join('\n')
    : 'N/A';
};

function CategorySummary({ results, category }) {
  const filteredResults = useMemo(() => results.filter((r) => r.Category === category), [results, category]);

  return <p>{getFormattedResult(category, filteredResults)}</p>;
}
```

Issues resolved:
- Mutation of array in `Array.sort`
- Only the last entity gets used for medical conditions and medications
- Extra newline characters at the end of lists
- `formattedResult` can either be a string or array based on the conditional logic
- Extra `div` being rendered
- Overall readability (`Array.map` over `for` loop, template syntax and `Array.join` over string concatenation, arrow function over `function`)

Reply:
Thanks. Will update.

Regarding your comment about the selected concept, I have added that comment [here](https://github.com/aws-samples/medical-transcription-analysis/pull/55#discussion_r589827947)